azure-spring-boot/src/main/java/com/microsoft/azure/spring/autoconfigure/b2c/AADB2CEntryPoint.java

```diff
@@ -41,9 +41,8 @@ public void commence(HttpServletRequest request, HttpServletResponse response, A
         }
 
         final String redirectURL = AADB2CURL.getOpenIdSignUpOrSignInURL(b2cProperties, requestURL, request);
+        redirectStrategy.sendRedirect(request, response, redirectURL);
 
         log.debug("Authentication is required to access URL {}.", requestURL);
-
-        redirectStrategy.sendRedirect(request, response, redirectURL);
```

Comment:
Why do we change the order here?

Reply:
Yes, should not change this, will revert this.

Reply:
Also let's refine the debug log to `Authentication is required to access URL {}. Redirecting to {}`.